base/types/gfsperrors/error.go

```diff
@@ -56,20 +59,32 @@ func MakeGfSpError(err error) *GfSpError {
 	if err == nil {
 		return nil
 	}
-	switch e := err.(type) {
-	case *GfSpError:
-		if e.GetInnerCode() == 0 {
+
+	if gfspErr, ok := err.(*GfSpError); ok {
+		if gfspErr.GetInnerCode() == 0 {
 			return nil
 		}
-		return e
-	default:
-		return &GfSpError{
-			CodeSpace:      DefaultCodeSpace,
-			HttpStatusCode: int32(http.StatusInternalServerError),
-			InnerCode:      int32(DefaultInnerCode),
-			Description:    err.Error(),
+		return gfspErr
+	}
+	// Attempting to check if the error is an RPC error; if so, trying to extract its error description and convert it to the GfSpError type.
+	// Expected error should be: "rpc error: code = Unknown desc = xxx"
+	// We only need to focus on xxx, and can extract the substring using string.Index().
+	i := strings.Index(err.Error(), "desc = ")
+	if i != -1 && len(err.Error())-1 >= i+6 {
+		errInfo := err.Error()[i+6:]
+		var gfspErr GfSpError
+		unmarshalErr := json.Unmarshal([]byte(errInfo), &gfspErr)
+		if unmarshalErr == nil && gfspErr.HttpStatusCode != 0 {
```

Comment:
What does the 0 here mean?  200 means http is successful.

Reply:
0 indicates that err information is successfully resolved

Reply:
Why not use `gfspErr.InnerCode != 0`? I think `gfspErr.HttpStatusCode` should be 200 by default.